src/login/logind-session.c

```diff
@@ -87,6 +87,19 @@ int session_new(Session **ret, Manager *m, const char *id) {
         return 0;
 }
 
+static void session_reset_leader(Session *s) {
+        _cleanup_(pidref_freep) PidRef *pid_old = NULL;
+
+        assert(s);
+
+        if (!pidref_is_set(&s->leader))
+                return;
+
+        assert_se(hashmap_remove2(s->manager->sessions_by_leader, &s->leader, (void**) &pid_old) == s);
+
+        return pidref_done(&s->leader);
```

Comment:
In this stage, `s->leader` and `pid_old` points to the same object, right? Hmm...